ports/stm32/qspi.c

```diff
@@ -112,10 +112,19 @@ void qspi_init(void) {
         ;
 }
 
-void qspi_memory_map(void) {
+void qspi_memory_map() {
     // Enable memory-mapped mode
 
     QUADSPI->ABR = 0; // disable continuous read mode
+
+    uint8_t cmd = 0xeb; // quad read opcode
+    uint8_t adsize = 2; // 24-bit address size
+    // For chip addresses over 16MB, 32 bit addressing is required instead of 24 bit
+    if ((MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2 -3 -1) >= 24) {
```

Comment:
Since this is a macro this test could probably become static/compile-time, and this whole set of changes made compile-time configurable.

Although there could be value in keeping it dynamically selectable I think for now it'd be good to try and make it static.

Eg:
```c
#if (MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2 -3 -1) >= 24
#define QSPI_CMD 0xec
#define QSPI_ADSIZE 3
#else 
#define QSPI_CMD 0xeb
#define QSPI_ADSIZE 2
#endif 
```

Reply:
While I like the idea of getting to the point of being able to configure spi / qspi flash devices at runtime, this does make sense to have static in the mean time! It can be reverted to a dynamic form if/when the rest of the module supports it!